src/xo-app/vm/tab-disks.js

```diff
@@ -1,146 +1,481 @@
+    }
+  }
+
+  _reset = () => {
+    const { vm } = this.props
+    const order = this._parseBootOrder(vm.boot && vm.boot.order)
+    this.setState({order})
+  }
+
+  _save = () => {
+    const { vm, onClose = noop } = this.props
+    const { order: newOrder } = this.state
+    let order = ''
+    forEach(newOrder, item => { item.active && (order += item.id) })
+    return setVmBootOrder(vm, order)
+      .then(onClose)
+  }
+
+  render () {
+    const { order } = this.state
+
+    return <form>
+      <ul>
+        {map(order, (item, index) => <OrderItem
+          key={index}
+          index={index}
+          id={item.id}
+          item={item}
+          move={this.moveOrderItem}
+        />)}
+      </ul>
+      <fieldset className='form-inline'>
+        <span className='pull-right'>
+          <ActionButton icon='save' btnStyle='primary' handler={this._save}>Save</ActionButton>
+          {' '}
+          <ActionButton icon='reset' btnStyle='default' handler={this._reset}>Reset</ActionButton>
+        </span>
+      </fieldset>
+    </form>
+  }
+}
+
+export default class TabDisks extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      attachDisk: false,
+      bootOrder: false,
+      newDisk: false
+    }
+  }
+
+  _toggleNewDisk = () => this.setState({
+    newDisk: !this.state.newDisk,
+    attachDisk: false,
+    bootOrder: false
+  })
+
+  _toggleAttachDisk = () => this.setState({
+    attachDisk: !this.state.attachDisk,
+    bootOrder: false,
+    newDisk: false
+  })
+
+  _toggleBootOrder = () => this.setState({
+    bootOrder: !this.state.bootOrder,
+    attachDisk: false,
+    newDisk: false
+  })
+
+  render () {
+    const {
+      srs,
+      vbds,
+      vdis,
+      vm
+    } = this.props
+
+    const {
+      attachDisk,
+      bootOrder,
+      newDisk
+    } = this.state
+
+    return <Container>
+      <Row>
+        <Col className='text-xs-right'>
+          <TabButton
+            btnStyle={newDisk ? 'info' : 'primary'}
+            handler={this._toggleNewDisk}
+            icon='add'
+            labelId='vbdCreateDeviceButton'
+          />
+          <TabButton
+            btnStyle={attachDisk ? 'info' : 'primary'}
+            handler={this._toggleAttachDisk}
+            icon='disk'
+            labelId='vdiAttachDeviceButton'
+          />
+          <TabButton
+            btnStyle={bootOrder ? 'info' : 'primary'}
+            handler={this._toggleBootOrder} // TODO: boot order
+            icon='sort'
+            labelId='vdiBootOrder'
+          />
+        </Col>
+      </Row>
+      <Row>
+        {newDisk && <div><NewDisk vm={vm} vbds={vbds} vdis={vdis} onClose={this._toggleNewDisk} /><hr /></div>}
+        {attachDisk && <div><AttachDisk vm={vm} vbds={vbds} vdis={vdis} onClose={this._toggleAttachDisk} /><hr /></div>}
+        {bootOrder && <div><BootOrder vm={vm} onClose={this._toggleBootOrder} /><hr /></div>}
+      </Row>
+      <Row style={{ minWidth: '0' }}>
```

Comment:
I don't know, this is not my work, neither the Row/Col choice upward...I never chose to Use these Components by myself, so I'm not aware of any display issues when using them...